### PR TITLE
ENH Unvendor unit tests from packages

### PR DESCRIPTION
**Pull request by rth**
Closes https://github.com/pyodide/pyodide/issues/617

Partially addresses https://github.com/pyodide/pyodide/issues/646#issuecomment-649124358

This unvendors unit tests included in Python packages, and moves them to a separate package `<package-name>-tests`.  This results in around 20% size reduction on average for packages that vendor tests,

|              | Size in MB (main) | Size in MB (PR) | Size reduction (%) |
|--------------|-------------------|-----------------|--------------------|
| numpy        | 8.3               | 6.8             | 18                 |
| pandas       | 15                | 11              | 26                 |
| scipy        | 22                | 18              | 18                 |
| sympy        | 14                | 8.9             | 36                 |
| jedi         | 1.8               | 1.8             | 1                  |
| scikit-image | 28                | 20              | 28                 |
| scikit-learn | 9.3               | 7.6             | 18                 |
| statsmodels  | 12                | 4.6             | 61                 |

For packages that do not include any unit tests in the installation folder, `<package-name>-tests` would not exist. One can determine whether unit tests were unvendored by looking at the `unvendored_tests` field in `packages.json`.

### CODE REVIEW

File: docs/development/building-from-sources.md [hunk 1]
```diff
@@ -96,7 +96,8 @@ they are case sensitive.
 
 If `PYODIDE_PACKAGES` is not set, a minimal set of packages necessairy to run
 the core test suite is installed, including "micropip", "pyparsing", "pytz",
-"packaging", "Jinja2". This is equivalent to setting `PYODIDE_PACKAGES='core'`
+"packaging", "Jinja2", "regex". This is equivalent to setting
```
Comment (rth) [inline docs/development/building-from-sources.md:99]: I had to add it for testing, as none of the other had vendored tests. It's good it have a package that needs compilation in the core meta-package in any case though.

**Review by hoodmane — APPROVED** [review]
Looks good, thanks!
